hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/RequestHandler.java

```diff
@@ -132,15 +132,13 @@ private boolean syncIfLocalViewBehind(Context ctx) {
       String lastKnownInstantFromClient =
           ctx.queryParam(RemoteHoodieTableFileSystemView.LAST_INSTANT_TS, HoodieTimeline.INVALID_INSTANT_TS);
       SyncableFileSystemView view = viewManager.getFileSystemView(basePath);
-      synchronized (view) {
```

Comment:
hi, @chaplinthink I think it is necessary to use synchronization to sync view locally since the handler would handle different request from clients concurrently. cc @bvaradar

Reply:
I mean the implementation of view.sync(); already has WriteLock to handle multiple requests from clients concurrently

```
 try {
      writeLock.lock();
      runSync(oldTimeline, newTimeline);
    } finally {
      writeLock.unlock();
    }
```

Reply:
@chaplinthink Thanks for the explaination, make sense to me. @vinothchandar @bvaradar do you have any other concern?

Reply:
I was mulling about the reloading of timeline that happens before the write lock. 

```
@Override
  public void sync() {
    HoodieTimeline oldTimeline = getTimeline();
    HoodieTimeline newTimeline = metaClient.reloadActiveTimeline().filterCompletedAndCompactionInstants();
    try {
      writeLock.lock();
      runSync(oldTimeline, newTimeline);
    } finally {
      writeLock.unlock();
    }
  }
```

`runSync()` actually could init/reassign `metaClient`, so in theory removing synchornized could in theory make it non-serializable. 


I would suggest that we either move the timeline reload into the write lock and leave this as-is. Whatever we change, we need to validate with more concurrent testing. So not sure if this is all worth the trouble. 

Are you hitting real concurrency bottlenecks around this?

Reply:
Thanks a lot for reply.  Do you mean the  `synchornized `  is to ensure  `HoodieTimeline newTimeline = metaClient.reloadActiveTimeline().filterCompletedAndCompactionInstants();`  concurrently?

In fact, we are also to do this right ?

```
@Override
  public void sync() {
    HoodieTimeline oldTimeline = getTimeline();
    try {
      writeLock.lock();
      HoodieTimeline newTimeline = metaClient.reloadActiveTimeline().filterCompletedAndCompactionInstants();
      runSync(oldTimeline, newTimeline);
    } finally {
      writeLock.unlock();
    }
  }
```

I am confused when I see the code that we use  `synchornized`  and  `writeLock`  at the same time.

I agree to validate this with more concurrent testing. Currently i have not encountered  concurrency bottlenecks.